back/engines/commercial/insights/app/services/insights/views/create_service.rb

```diff
@@ -9,8 +9,15 @@ class CreateService
 
       def initialize(current_user, params)
         @current_user = current_user
-        @params = params.dup
-        @params[:data_sources_attributes] = @params.delete(:data_sources)
+
+        @params = params.dup.to_h.tap do |p|
+          data_sources = p.delete(:data_sources)
+          raise ArgumentError, 'At least one data source must be provided' if data_sources.blank?
+
+          # Since Project is the only supported origin_type for now, we use it as the
+          # default and make the origin_type attribute optional.
+          p[:data_sources_attributes] = data_sources.map { |ds| { 'origin_type' => 'Project', **ds } }
```

Comment:
```suggestion
          p[:data_sources_attributes] = data_sources.map { |ds| ds.merge({ 'origin_type' => 'Project' }) }
```

I'd prefer to use `Map#merge` instead of the double splatter for clarity, but not a biggie. :+1:

Reply:
A bit opiniated, I guess 🙂 In that case, this should be something like
```ruby
{'origin_type' => 'Project'}.merge(ds)
```
otherwise we would override the origin_type instead of setting a default value.